src/android/com/microsoftopentechnologies/intellij/wizards/createvm/SelectImageStep.java

```diff
@@ -0,0 +1,312 @@
+            @Override
+            public void itemStateChanged(ItemEvent itemEvent) {
+                fillList();
+            }
+        });
+
+        imageLabelList.setCellRenderer(new DefaultListCellRenderer() {
+            @Override
+            public Component getListCellRendererComponent(JList<?> jList, Object o, int i, boolean b, boolean b1) {
+                String cellValue = (o instanceof VirtualMachineImage) ? ((VirtualMachineImage) o).getLabel() : o.toString();
+                this.setToolTipText(cellValue);
+                return super.getListCellRendererComponent(jList, cellValue, i, b, b1);
+            }
+        });
+
+        imageLabelList.addListSelectionListener(new ListSelectionListener() {
+            @Override
+            public void valueChanged(ListSelectionEvent listSelectionEvent) {
+                VirtualMachineImage virtualMachineImage = (VirtualMachineImage) imageLabelList.getSelectedValue();
+                model.setVirtualMachineImage(virtualMachineImage);
+
+                if(virtualMachineImage != null) {
+                    imageDescriptionTextPane.setText(model.getHtmlFromVMImage(virtualMachineImage));
+                    imageDescriptionTextPane.setCaretPosition(0);
+                    model.getCurrentNavigationState().NEXT.setEnabled(true);
+                }
+            }
+        });
+
+        imageDescriptionTextPane.addHyperlinkListener(new HyperlinkListener() {
+            @Override
+            public void hyperlinkUpdate(HyperlinkEvent hyperlinkEvent) {
+                if(hyperlinkEvent.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
+                    if(Desktop.isDesktopSupported()) {
+                        try {
+                            Desktop.getDesktop().browse(hyperlinkEvent.getURL().toURI());
+                        } catch (Exception e) {
+                            UIHelper.showException("Error opening link", e);
+                        }
+                    }
+                }
+            }
+        });
+    }
+
+    @Override
+    public JComponent prepare(WizardNavigationState wizardNavigationState) {
+        rootPanel.revalidate();
+
+        if(virtualMachineImages == null) {
+            imageTypeComboBox.setEnabled(false);
+            model.getCurrentNavigationState().NEXT.setEnabled(false);
+
+            imageLabelList.setListData(new String[]{"loading..."});
+            imageLabelList.setEnabled(false);
+
+
+            ProgressManager.getInstance().run(new Task.Backgroundable(project, "Loading virtual machine images...", false) {
+                @Override
+                public void run(ProgressIndicator progressIndicator) {
+                    progressIndicator.setIndeterminate(true);
+
+                    try {
+                        for (VirtualMachineImage virtualMachineImage : AzureSDKManagerImpl.getManager().getVirtualMachineImages(model.getSubscription().getId().toString())) {
```

Comment:
Should we cache the result of calling `getVirtualMachineImages`? I am just thinking of the case when the user hits the **Previous** button.